scene/3d/audio_stream_player_3d.cpp

```diff
@@ -595,83 +543,49 @@ void AudioStreamPlayer3D::seek(float p_seconds) {
 
 void AudioStreamPlayer3D::stop() {
 	setplay.set(-1);
-	for (Ref<AudioStreamPlayback> &playback : stream_playbacks) {
+	internal.stop();
+	for (Ref<AudioStreamPlayback> &playback : internal.stream_playbacks) {
 		AudioServer::get_singleton()->stop_playback_stream(playback);
 	}
-	stream_playbacks.clear();
-	active.clear();
-	set_physics_process_internal(false);
 }
 
 bool AudioStreamPlayer3D::is_playing() const {
-	for (const Ref<AudioStreamPlayback> &playback : stream_playbacks) {
-		if (AudioServer::get_singleton()->is_playback_active(playback)) {
-			return true;
-		}
-	}
 	if (setplay.get() >= 0) {
 		return true; // play() has been called this frame, but no playback exists just yet.
 	}
-	return false;
+	return internal.is_playing();
 }
 
 float AudioStreamPlayer3D::get_playback_position() {
-	// Return the playback position of the most recently started playback stream.
-	if (!stream_playbacks.is_empty()) {
-		return AudioServer::get_singleton()->get_playback_position(stream_playbacks[stream_playbacks.size() - 1]);
-	}
-	return 0;
+	return internal.get_playback_position();
 }
 
 void AudioStreamPlayer3D::set_bus(const StringName &p_bus) {
-	//if audio is active, must lock this
-	AudioServer::get_singleton()->lock();
-	bus = p_bus;
-	AudioServer::get_singleton()->unlock();
+	internal.bus = p_bus;
```

Comment:
```suggestion
	internal.bus = p_bus; // This will be pushed to the audio server during the next physics timestep, which is fast enough.
```

And could you add this info to the AudioStreamPlayer2D/3D docs, please?

Reply:
Somewhat out of scope of the PR since it's always been the case, pretty much

Reply:
The comment was copied from same func from AudioStreamPlayer2D.

The docs update is just for info for users to not be confused about possible delays if their physics process is stuttering

Reply:
I'd wait until a confused user opens an issue. So far it hasn't been a problem to anyone, likely because changing buses dynamically is a very uncommon action.
I copied the comment though.